### PR TITLE
Update `random` module and fix some issues

### PR DESCRIPTION
**Pull request by Snowapril**
This revision do follow things
*  Update `random.py` and `test_random.py` to cpython 3.10
*  Fix `os.urandom` and `random.getrandbits` to check input sign
*  Remove `random.randbytes` as it is already on `Random` class in `random.py` and python-side code never reach there

### CODE REVIEW

**Comment by Snowapril** [comment]
It seems `unic_ucd_category` crate makes miri test failed

**Comment by fanninpm** [comment]
Now that #3521 is merged, would you please consider rebasing this branch onto a fresh copy of `main`?

**Comment by Snowapril** [comment]
@fanninpm Oh sorry for being late. I'll rebase it tonight 😊

File: stdlib/src/random.rs [hunk 1]
```diff
@@ -110,46 +110,42 @@ mod _random {
...
-                    r >> (32 - k)
-                } else {
-                    r
+            match k {
+                k if k < 0 => {
+                    Err(vm.new_value_error("number of bits must be non-negative".to_owned()))
+                }
+                0 => Ok(BigInt::zero()),
+                _ => {
+                    let mut rng = self.rng.lock();
```
Comment (fanninpm) [inline stdlib/src/random.rs:119]: This feels much more elegant than what I wrote. Thanks!

File: Lib/random.py [hunk 2]
```diff
@@ -48,25 +48,13 @@
...
 from _collections_abc import Set as _Set, Sequence as _Sequence
+from operator import index as _index
 from itertools import accumulate as _accumulate, repeat as _repeat
 from bisect import bisect as _bisect
-try:
-    import os as _os
-except ImportError:
-    # XXX RUSTPYTHON
-    # On wasm, we don't have the os module
-    _os = None
```
Comment (coolreader18) [inline Lib/random.py:69]: Can you make sure to keep these try blocks? They keep `random` working on wasm.

**Comment by coolreader18** [comment]
Sorry this took so long to review; my second semester started recently and it was discouraging looking at stuff when CI was permanently red :grimacing:

File: Lib/random.py (hunk 2 above)
Comment (youknowone) [reply Lib/random.py:69]: @Snowapril could you check this comment?

Comment (Snowapril) [reply Lib/random.py:69]: Oh, I'm sorry noticed so late 😭. I rebased it.

**Review by youknowone — APPROVED** [review]

**Pull request merged**